lib/core/src/Cardano/Wallet.hs

```diff
@@ -718,7 +719,9 @@ walletSyncProgress ctx w = do
     (_,_,st) = ctx ^. genesisData
 
     ti :: TimeInterpreter IO
-    ti = timeInterpreter (ctx ^. networkLayer @t)
+    ti = neverFails
```

Comment:
This always felt a bit wrong — to motivate the `neverFails` _outside_ the `syncProgress` function, with `syncProgress` internals as arguments — but I wasn't sure how to fix.

We could make `syncProgress` take a `TimeInterpreter ExcepT PastHorizonException IO` and return `IO SyncProgress`, but then we couldn't use it with `TimeInterpreter Identity` …

Reply:
Although `walletSyncProgress` "only converts times at the tip", there's no guarantee that the tip itself isn't on an unsafe zone, isn't it?

Reply:
The safe-zone is at least a significant portion of an epoch. And we expect the wallet tip to be behind the node tip.